packages/design-system/src/components/Table/Table.jsx

```diff
@@ -17,108 +16,118 @@ function debounce(fn, ms) {
+
+    if (process.env.NODE_ENV !== 'production') {
+      if (
+        props.scrollable &&
+        Array.isArray(props.children) &&
+        !props.children.some((child) => child.type === TableCaption)
+      ) {
+        console.warn(
+          'The children prop in `Table` must include `TableCaption` component for scrollable tables.'
+        );
+      }
     }
   }
 
-  const container = useRef(null);
-  // The captionID is stored as init value of a ref.
-  const captionID = useRef(uniqueId('caption-'));
-  const [isTableScrollable, setTableScrollable] = useState(false);
-
-  // Listens to the window resize event to dynamically handle horizontal scrollable tables
-  useEffect(() => {
-    function checkScrollable() {
-      const { scrollWidth, clientWidth } = container.current;
-      const isScrollActive = scrollWidth > clientWidth;
-      setTableScrollable(isScrollActive);
+  componentDidMount() {
+    if (this.props.scrollable) {
+      window.addEventListener('resize', this.debounceHandleResize);
+      this.handleResize();
     }
+  }
 
-    if (scrollable) {
-      checkScrollable();
-
-      // Set isTableScrollable `true` if the table width is wider than the viewport
-      const debouncedHandleResize = debounce(function handleResize() {
-        checkScrollable();
-      }, 500);
-
-      // Create window resize event listener that calls the debouncedHandleResize function
-      window.addEventListener('resize', debouncedHandleResize);
-
-      // Clean up the establishd event listeners
-      return () => {
-        window.removeEventListener('resize', debouncedHandleResize);
-      };
+  componentWillUnmount() {
+    if (this.props.scrollable) {
+      window.removeEventListener('resize', this.debounceHandleResize);
     }
-  }, [scrollable]);
-
-  const classes = classNames(
-    'ds-c-table',
-    striped ? 'ds-c-table--striped' : null,
-    stackBreakpoint ? `ds-c-table--stacked-${stackBreakpoint}` : null,
-    className
-  );
-
-  // Make table container focusable and display scroll notice when table width exceeds viewport.
-  // Set attribute `tabIndex = 0` makes table container focusable enables keyboard support of using the arrow keys.
-  // Also, it provides context for screen reader users as they are able to focus on the region.
-  // Do this by using table's <caption> to label the scrollable region using aria-labelleby
-  const attributeScrollable = scrollable && {
-    className: 'ds-c-table__wrapper',
-    role: 'region',
-    'aria-labelledby': captionID.current,
-    'aria-live': 'polite',
-    'aria-relevant': 'additions',
-    tabIndex: isTableScrollable ? '0' : null,
-  };
+  }
 
-  const isTableCaptionComponent = (child) => {
-    return child && child.type === TableCaption;
-  };
+  handleResize() {
+    const { scrollWidth, clientWidth } = this.container;
+    const scrollActive = scrollWidth > clientWidth;
+    this.setState({
+      tabIndex: scrollActive ? '0' : null,
```

Comment:
@nichia It looks like this doesnt incorporate later feedback in the original table PR, i.e. using `isTableScrollable` instead of `tabIndex`. If the changes in this PR are just made by reverting to old commits, can you make sure to incorporate all the later updates as well?

Reply:
Ah, will add back